src/qibojit/backends/matrices.py

```diff
@@ -4,6 +4,15 @@
 from qibo.backends.npmatrices import NumpyMatrices
 
 
+class CupyMatrices(NumpyMatrices):  # pragma: no cover
+    def __init__(self, dtype):
+        super().__init__(dtype)
+        import cupy as cp  # pylint: disable=import-error
+
+        # Define matrices by simply replacing np with cp in NumpyMatrices
+        self.np = cp
```

Comment:
I don't know if this is the best solution but it seems to work.
Perhaps we could improve readability by changing the name of the attribute `np`, even if I know that for specify methods we might need to access numpy also in backends different from the `NumpyBackend`. So I'm not sure.

Reply:
To me it is fine, because I think if you change its name it will break `TensorflowMatrices`. But I'd still confirm that we really need the `CupyMatrices` before thinking about this because maybe we don't need to add this at all.